### PR TITLE
Upgrade to Rails 4.2.4

### PR DESCRIPTION
**Pull request by boffbowsh**
Best reviewed commit by commit.

Highlights:

- Convert everything to ActiveSupport test cases
- Routing and rendering changes for print formats, now uses "variants"
- Added govuk-lint in diff mode

### CODE REVIEW

**Comment by jamiecobbett** [comment]
Related: https://github.gds/gds/alphagov-deployment/pull/1042

File: Gemfile [hunk 1]
```diff
@@ -33,19 +33,9 @@ gem 'logstasher', '0.4.8'
...
-group :assets do
-  # Note that govuk_frontend_toolkit is only used here for SASS mixins and
-  # variables. Analytics and other javascript features are provided by static.
-  gem 'govuk_frontend_toolkit', '~> 4.1.0'
-  gem 'sass', "3.3.14"
-  gem 'sass-rails', "3.2.6"
-  gem "therubyracer", "0.12.2"
-  gem 'uglifier'
-end
+gem 'govuk-lint'
```
Comment (benilovj) [inline Gemfile]: This probably shouldn't be a runtime gem

Comment (boffbowsh) [inline Gemfile]: Updated.

**Comment by benilovj** [comment]
:+1:

**Pull request merged**